e2e-tests/src/partner_chain_rpc.py

```diff
@@ -47,51 +47,51 @@ def __get_body(self, jsonrpc="2.0", method="", params=[], id=1):
         }
 
     def __exec_rpc(self, method: str, params: list = []):
-        """
-        Execute an RPC request either via HTTP or kubectl exec.
-        Returns the raw JSON response for further processing.
-        """
+        # Check if we should use kubectl exec and if required env vars are set
+        use_kubectl = os.environ.get("USE_KUBECTL_RPC") == "true"
+        kubectl_pod_exists = "KUBECTL_EXEC_POD" in os.environ
+        
+        # If kubectl mode is desired but required env var is missing, log warning and fall back
+        if use_kubectl and not kubectl_pod_exists:
+            logger.warning("USE_KUBECTL_RPC is set to true but KUBECTL_EXEC_POD is not defined. Falling back to HTTP request.")
+            use_kubectl = False
+        
+        # Use the original HTTP method implementation
+        if not use_kubectl:
+            response = requests.post(
+                self.url,
+                headers=self.headers,
+                json=self.__get_body(method=method, params=params)
+            )
+            return response.json()
+        
+        # Use kubectl exec method
+        pod = os.environ["KUBECTL_EXEC_POD"]
+        namespace = os.environ.get("K8S_NAMESPACE", "default")
+        
+        payload = json.dumps(self.__get_body(method=method, params=params))
+        cmd = [
+            "kubectl", "exec", pod, "-n", namespace, "--",
+            "curl", "-s", "-H", "Content-Type: application/json",
+            "-d", payload, "http://localhost:9933"
```

Comment:
```suggestion
            "-d", payload, self.url
```
I would change that. Users can select which node they want to connect to/test in config:
```
"selected_node": "validator-4",
```
and in future we want to test random nodes, so I’d like to keep it that way. It should be as easy as updating hosts in `<env>-ci.json`. I assume fqdn that we can only use within aws network are no longer be needed since we're switching to kubectl.